### ISSUE TITLE
[Plugin] linkerd

### ISSUE THREAD

**Issue by benjdlambert** [spotify/backstage#633]
## Summary

- Be able to see the linkerd cluster that is currently running
- Get real time reports of traffic running through the system
- Be able to map those to components in Backstage
- Build a network topology of your entire stack and how components talk to other components.

## Project website (if applicable)
https://linkerd.io/

## Context
Ultralight, security-first service mesh for Kubernetes
Linkerd adds critical security, observability, and reliability features to your Kubernetes stack—no code change required.

**Comment by benjdlambert** [comment]
Parallel issue also opened on https://github.com/linkerd/linkerd2/issues/4955,

If anyone is interested to help make this happen part of hacktoberfest would be super cool!

**Comment by Pothulapati** [comment]
I'm one of the Maintainers of the Linkerd project. :wave: The team is more than happy to support this effort in making this happen as part of the Hacktoberfest. 

If someone is interested to take this up, Feel free to join the `contributors` channel in the [Linkerd Slack](https://slack.linkerd.io/) and ask any questions. Also, this Issue and https://github.com/linkerd/linkerd2/issues/4955 are great places to continue the conversation.

**Comment by benjdlambert** [comment]
Still ongoing!

**Comment by samiurrehman92** [comment]
Hey guys, how is it going? cc @benjdlambert

**Comment by adamdmharvey** [comment]
I do not believe Ben is working on this plugin; you'll see it remains unassigned. I believe he was simply suggesting the stale bot should not have closed it, so he reopened it.

I'll offer that this is an open issue (with no progress yet) that anyone from the community is welcome to engage on and support!  If you'd like to work on it, let the group know and we'll assign it to you!  Love to see what the community can come up with!

**Comment by benjdlambert** [comment]
@adamdmharvey I know it looks that way, but I did spend a little time in building this. Unfortunately i've not had so much time to spend on this as I'd hoped as it just falls under hackdays or weekend time right now for me as it's not something we'd prioritise part of core maintainer work. 

I can give a little update however from the last time I worked on this.

The code is in still a little bit of a hacky/exploration stage in places, so it's pretty far off being anything like production ready. I managed to map backstage entities to deployments and services and using https://reactflow.dev/ building somethink like the octopus graph for upstream and downstream dependencies.

If anyone want's to take over this work i'd be more than happy to share what i've done so far, or if someone want's to help out, feel free to reach out!

I'm hoping to maybe get some free time a over the summer and try and get this to an alpha release stage so that we can start to use some of the linkerd data in backstage 👍

**Comment by samiurrehman92** [comment]
Thanks for sharing @benjdlambert. Our team is still just in early adoption phase for both `linkerd` and `backstage`, since they were happening both at the same time, I was curious about the effort going into this plugin. Once we've adopted both tools, we will naturally consider contributing toward this plugin and reach out then.

**Comment by ehrnst** [comment]
Hi. Is there any active development on Linkerd support for backstage?

**Comment by benjdlambert** [comment]
@ehrnst hey :wave: so long story short is that this was something I picked up as a spare time thing, and then Backstage exploded so not really had much time to look at finishing it off. Is it something that you would like to help build?

:pray:

**Comment by ehrnst** [comment]
I would really like a feature where we could identify if documented dependencies are different from what linkerd sees. However, I do not have any skills required for doing any coding here ☹️

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
